### ISSUE TITLE
unref lock check interval

### ISSUE THREAD

**Issue by bmeck** [IndigoUnited/node-proper-lockfile#3]
I believe that lock files have no need for keeping a node process alive if there is no work being done w/ them. Is there any reason we cannot unref the timer?

**Comment by satazor** [comment]
The timer only starts when you acquire the lock and is stopped when you release it. Why aren't you releasing the lock?

**Comment by bmeck** [comment]
this was found when I had a bug in my code actually. we are pulling work
off a queue and the work became abandoned due to a bug. however, we were
very confused as to why the process was kept alive when normally if no work
is left to be done it would die.

**Comment by satazor** [comment]
Well thats how proper lock file prevents staleness of long running processes: it updates the lock file at a given interval. Can I close this?

**Comment by bmeck** [comment]
the question is if you have abandoned locks why are they keeping a process open

IE: why does the following not exit

```javascript
#!/usr/bin/env node
require('proper-lockfile').lock('package.json', {}, function () {})
```

**Comment by marcooliveira** [comment]
You have an open lock on `package.json`. You need to release it before the process exits. Underneath, as @satazor pointed out, there is a timer running, which prevents the process from exiting. Regardless of the fact that you are providing an empty callback to `lock`, there is no way of telling if there is anything else running on your process. The API is simple. You open a lock, it remains locked until you unlock it.

**Comment by bmeck** [comment]
Actually node use reference counting to see if any work is pending, it
automatically will close when no work is left to do, this is why

```
setTimeout(function(){})
```

closes after the timeout handle `unref`s. because of this there is no need
to test if there is more work queued, as node would try to exit when there
is no more work queued automatically.

**Comment by marcooliveira** [comment]
As you correctly point out, node does keep an eye out for pending work, which is why the setInterval inside is keep node alive. There is no way of knowing, from inside the module, if there is nothing else pending and thus stopping the interval so that node dies. Just release the lock, and you're done.

**Comment by bmeck** [comment]
@marcooliveira using timer.unref() you can mark the interval as not being counted for keeping node alive, there is no reason it should be counted. there is no need to check if there is more work.

**Comment by satazor** [comment]
I didnt even know about timer.unref.. Its ok if we do that, because this library will remove any pending locks on process exit. @bmeck can you do the changes?

**Comment by marcooliveira** [comment]
Uh! Nice one. Thanks for the contribution! Was also not aware of that.

**Comment by bmeck** [comment]
@satazor made the pr, can't think of a non-racey/non-unsupported way to test this really though

**Comment by satazor** [comment]
Add tests for it here: https://github.com/IndigoUnited/node-proper-lockfile/commit/ec3b960440948ce17292b180789d8490e0111e53

**Comment by satazor** [comment]
Changed landed in `0.6.1`
